websocket_server.go

```diff
@@ -346,15 +346,29 @@ func (n *pushServer) start(ctx context.Context) {
 					}
 
 					if n.cfg.pushServer.dispatchHandler != nil {
-						ok, err := n.cfg.pushServer.dispatchHandler.ShouldDispatch(session, event, eventSummary)
+						dispatch, err := n.cfg.pushServer.dispatchHandler.ShouldDispatch(session, event, eventSummary)
 						if err != nil {
+							// if 'ShouldDispatch' returns an error type that has the method 'ShouldCloseSocket() (bool, int)'
+							// then it can inform whether the socket should be closed along with a code to use in the
+							// close message. This is useful in cases where it makes no sense to continue serving the client
+							// (e.g. the client configured a semantically invalid push config, for example: they used a comparator
+							// that is not yet supported/implemented).
+							if closer, ok := err.(interface {
+								ShouldCloseSocket() (bool, int)
+							}); ok {
+								if close, code := closer.ShouldCloseSocket(); close {
+									session.close(code)
+									session.unregister(session)
```

Comment:
I am not certain if this is necessary, but figured it wouldn't hurt removing it from the `sessions` slice so we don't iterate over it again.

@primalmotion thoughts?

Reply:
I think that close() will call unregister. can you verify, and if so remove the call from here?

Reply:
I can confirm that `close` does not call `unregister`.